projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php

```diff
@@ -153,7 +153,7 @@ public function access_url( $mode = 'subscribe' ) {
 			$permalink = add_query_arg( $wp->query_vars, home_url( $wp->request ) );
 		}
 
-		$login_url = $this->get_rest_api_token_url( $this->get_site_id(), $permalink );
+		$login_url = $this->get_rest_api_token_url( get_current_blog_id(), $permalink );
```

Comment:
This change is problematic because on Jetpack `get_current_blog_id` has a different meaning and we want to get the WPCOM blog id.
Jetpack site may be a multisite as well and you will just get the blog id of the Jetpack site here

Reply:
I don't understand that comment...

> on Jetpack get_current_blog_id has a different meaning and we want to get the WPCOM blog id.

What's the "Jetpack meaning"?

At the end of the day, `$this->get_site_id()` was calling `get_current_blog_id()` so there is no difference here, is there?

Reply:
Jetpack stores the id of the connected shadow site as an option. Because that ID refers to WPCOM id

Reply:
I see!!!